lib/src/body.dart

```diff
@@ -52,9 +52,13 @@ class Body {
         contentLength = encoded.length;
         stream = Stream.fromIterable([encoded]);
       }
+    } else if (body is List<int>) {
+      // Avoid performance overhead from an unnecessary cast.
+      contentLength = body.length;
+      stream = Stream.value(body);
     } else if (body is List) {
       contentLength = body.length;
-      stream = Stream.fromIterable([body.cast()]);
+      stream = Stream.value(body.cast());
     } else if (body is Stream) {
```

Comment:
I wonder, while we are here should we assume that `Stream.cast` is likely to have similar overhead?

```suggestion
    } else if (body is Stream<int>) {
      stream = body;
    } else if (body is Stream) {
```

Reply:
Definitely, seems worth the optimization